### ISSUE TITLE
Add support for more compression algorithms in File source

### ISSUE THREAD

**Issue by zamazan4ik** [vectordotdev/vector#16891]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Right now Vector supports reading files compressed only with GZip - https://vector.dev/docs/reference/configuration/sources/file/#compressed-files . But other algorithms (like Zstd or Snappy) could be used nowadays to store compressed logs.

### Attempted Solutions

Try to unpack logs externally before passing them to Vector but it's too inconvenient solution.

### Proposal

Add more compression algorithms like Zstd, Snappy, Bzip, etc.

**Comment by jszwedko** [comment]
@zamazan4ik do you have a specific compression algorithm you are interested in seeing support for? I think we should have one issue per algorithm.

**Comment by jszwedko** [comment]
Thanks @zamazan4ik ! I'll repurpose this one for zstd since I think that is likely to be most popular. We can let users open issues for the others with their use-cases.

**Comment by zamazan4ik** [comment]
In my practice, additionally to GZip, I've seen logs in Zstd and Bzip formats, stored in files. Never seen before logs in Snappy but could be useful to support it as well.

**Comment by dermetfan** [comment]
This looks like a duplicate of #13500

**Comment by jszwedko** [comment]
Agreed, I'll close the other one since this is more focused. Thanks for flagging @dermetfan

**Comment by mutexed** [comment]
Hello, any updates in this ticket? Interested in zstd compression for file source, may be there is some workaround?

**Comment by kvlas** [comment]
Any update? I'm also interested in zstd for file source.
